lib/index.ts

```diff
@@ -10,3 +15,290 @@ export enum Type {
     SHARE_TYPE_ROOM = 10,
     SHARE_TYPE_DECK = 12,
 }
+
+/**
+ * Successful OCS responses look like this, `T` is a template for the response data inside the response
+ */
+export interface NcOCSResponse<T>{
+    ocs: {
+        data: T,
+        meta: {
+            message: string,
+            status: string,
+            statuscode: number,
+        }
+    }
+}
+
+/**
+ * interface, generated using QuickType from response object
+ * TODO: find out where this is defined and make better
+ */
+export interface ShareData {
```

Comment:
No sure about this, maybe instead like the `Folder` or `Node` class of nextcloud-files?